### DISCUSSION TITLE
pyhanko.cli - ERROR - Error raised while producing signed file: The signer's certificate could not be validated.

### DISCUSSION THREAD

**Discussion by sayak02** [MatthiasValvekens/pyHanko#218] [Q&A]
**Describe the bug**
Adding custom signature to a pdf works in my local Node.js environment, but gives the "The Signer's certificate could not be validated error" in my deployed development environment in ECS.

Node.js Pyhanko Execution Command : 
`let command =
        "pyhanko sign addsig --field Signature_Of_" +
        user.name.replace(/\s/g, "_") +
        " --name " +
        user.name.replace(/\s/g, "_") +
        " --reason Agreement_to_this_Document_By_" +
        user.name.replace(/\s/g, "_") +
        " --location Sama_IN --with-validation-info --validation-context special-setup --use-pades pkcs12 " +
        pdfPath +
        " " +
        signedpath +
        "  file.pfx --passfil password.txt"`

where 

Attaching Docker File Config Screenshot for Reference: 
![Screenshot 2023-02-06 at 11 21 39 PM](https://user-images.githubusercontent.com/57043063/217047511-fe206629-2057-451d-abca-c567dd38ad88.png)

Attaching the pyhanko.yml config file
![Screenshot 2023-02-07 at 1 44 02 AM](https://user-images.githubusercontent.com/57043063/217075351-3b0a2a6b-c443-4a0b-845e-2609128bbda6.png)

Local Environment Python Version : 
python3 --version
Python 3.8.9

Pyhanko version 
pyHanko, version 0.13.2

**Comment by MatthiasValvekens** [comment]
Hi @sayak02 

Thanks for checking in. This type of validation error can be due to any number of factors. The most common culprits are the following:

 - Different trust settings between different environments. Are you sure that your environment actually has that `pyhanko.yml` config file and the `trust.txt`, `others.txt`,... files it refers to?
 - Network issues between different environments causing certain revocation info to become unavailable.


Hope that helps. If not, please post a full stack trace and all pyHanko-related log events.


(I'm moving this to the discussion board as it's a support question and not a bug report.)

**Reply by sayak02** [reply]
Hi @MatthiasValvekens ,
 I have currently resolved the issue by removing ` --with-validation-info` parameter from the command. This works just fine but currently unable to understand how that affects the whole esigning process.
